ouroboros-network-protocols/src/Ouroboros/Network/Protocol/LocalTxMonitor/Type.hs

```diff
@@ -121,6 +125,30 @@ data MempoolSizeAndCapacity = MempoolSizeAndCapacity
     -- ^ The number of transactions in the mempool
   } deriving (Generic, Eq, Show, NFData)
 
+data SizeAndCapacity a = SizeAndCapacity
+  { size     :: !a
+  , capacity :: !a
+  } deriving (Generic, Eq, Show, NFData)
+
+instance Functor SizeAndCapacity where
+  fmap f (SizeAndCapacity s c) = SizeAndCapacity (f s) (f c)
+
+data MeasureName
+  = TransactionBytes
+  | ExUnitsMemory
+  | ExUnitsSteps
+  | ReferenceScriptsBytes
+  | MeasureNameFromFuture !UnknownMeasureName
```

Comment:
What about defining
```haskell
newtype MeasureName = MeasureName Text
```
and then adding appropriate pattern synonyms in `ouroboros-consensus-cardano`?

I think this would have the following advantages:
 
 - `ouroboros-network` doesn't have to know about the Cardano measures, which are a ledger-specific thing, and it seems nice to keep `ouroboros-network` and `cardano-ledger` mostly decoupled (e.g. right now, `ouroboros-network` doesn't depend on any ledger packages).
 - The current illegal state `MeasureNameFromFuture (UnknownMeasureName "transaction_bytes")` gets irrepresentable.

WDYT?

Reply:
yep, that makes a lot of sense! I'll change it now 👍